astropy/io/ascii/cds.py

```diff
@@ -647,6 +650,32 @@ def write(self, lines):
 
                 to_pop.append(i)   # Delete original ``SkyCoord`` column.
 
+            # If the whole column is a `Time` or if it contains `Time` values,
+            # convert values to Modified Julian Dates.
+            elif isinstance(col, Time) or isinstance(col[0], Time):
+                # If column already has a description, then append to it.
+                name, description = 'MJD', 'Modified Julian Date'
+                if isinstance(col, Column):
+                    if col.description is not None:
+                        description = col.description + ' (Modified Julian Date)'
+                    if col.name is not None:
+                        name = col.name
+                try:
+                    self.cols[i] = Column([tval.mjd for tval in col],
+                                           name = name,
+                                           unit = u.day,
```

Comment:
This goes back to @taldcroft's comment 

> The u.day unit is still not appropriate, and probably worse than u.yr. The point here is that a Time does not have a unit. A value of 2019.0 is a decimal representation of a point in time that is equivalent to 2019-01-01, which is a string representation of that point in time. Neither of those have a meaningful unit.

This is applicable here too because Modified Julian Date is a "date" especially when we talk about it in terms of a `Time` column. `TimeDelta` would probably be more appropriate for using units, but I am not so sure about units being used here. However, this might be overly pedantic and maybe the MRT format doesn't talk about it. For now we can go ahead with this, but I will try to find out whether the format suggests a unit for Times, and why does it not ask for a `scale` since it is important for interpreting and round-tripping precise times.

Also, when the time scale is UTC, one day is not exactly 86400 seconds but `u.day` is.

Reply:
Sure. Will make the change. Thanks for the input. :)

Reply:
@aaryapatil became quite the expert in time systems in her GSOC, so we can trust here here!

Reply:
> This is applicable here too because Modified Julian Date is a "date" especially when we talk about it in terms of a `Time` column.

Ah, I recall having gone over while assigning the unit as `u.day`. The example on the MRT standard's page gives the unit of the BJD column as such. (https://journals.aas.org/mrt-standards/)

Again, I think the dates would have the unit of a _day_ since their resolution is a _day_, in spite of the fact that the defination of the _day_ may differ.
However, likewise can be argued for date value given in other formats and for the decimal year!?

cc: @taldcroft